elements/lisk-validator/src/keywords/data_type.ts

```diff
@@ -75,13 +80,40 @@ const compile = (
 		_parentDataProperty?: string | number,
 		_rootData?: object,
 	): boolean => {
-		if (value === 'boolean') return isBoolean(data);
-		if (value === 'bytes') return isBytes(data as Buffer);
-		if (value === 'string') return isString(data);
-		if (value === 'uint32') return isUInt32(data);
-		if (value === 'uint64') return isUInt64(data);
-		if (value === 'sint32') return isSInt32(data);
-		if (value === 'sint64') return isSInt64(data);
+		if (value === 'boolean') {
+			return isBoolean(data);
+		}
+		if (value === 'bytes') {
+			if (!isBytes(data as Buffer)) {
+				return false;
+			}
+			const parent = parentSchema as KVPair;
+			if (typeof parent.minLength === 'number') {
+				if ((data as Buffer).length < parent.minLength) {
+					return false;
```

Comment:
We have to check what error does this validation generates? A generic error like `dataType keyword error` will not guide user to troubleshoot what's wrong.

Reply:
Please add a test case for it as well.

Reply:
Test is here, but the return type expects boolean. How do we change the error message here?
https://github.com/LiskHQ/lisk-sdk/pull/5409/files#diff-712c76f4454b085e696bad5b60ddf1f8R175

Reply:
Change `errors: 'full'` and then you can return any custom error message. 

https://github.com/LiskHQ/lisk-sdk/blob/e13591fa7dcc18a93709eac8f9f60c0c6d0aae5f/elements/lisk-validator/src/keywords/data_type.ts#L125